src/app/shared/entry/recent-events.pipe.ts

```diff
@@ -25,27 +24,15 @@ export class RecentEventsPipe implements PipeTransform {
 
     switch (type) {
       case 'displayName': {
-        if (event.workflow) {
-          return this.entryToDisplayNamePipe.transform(event.workflow);
-        } else if (event.tool) {
-          return this.entryToDisplayNamePipe.transform(event.tool);
-        } else if (event.apptool) {
-          return this.entryToDisplayNamePipe.transform(event.apptool);
-        }
-        break;
+        const entry = this.getEntry(event);
+        return entry && this.entryToDisplayNamePipe.transform(entry);
       }
       case 'entryLink': {
-        if (event.workflow) {
-          return '/workflows/' + event.workflow.full_workflow_path;
-        } else if (event.tool) {
-          return '/containers/' + event.tool.tool_path;
-        } else if (event.apptool) {
-          return '/containers/' + event.apptool.full_workflow_path;
-        }
-        break;
+        const entry = this.getEntry(event);
+        return entry && '/' + entry.entryTypeMetadata.sitePath + '/' + this.getPath(entry);
       }
       case 'entryType': {
-        return event.tool || event.apptool ? this.EntryType.Tool : this.EntryType.BioWorkflow;
+        return this.getEntry(event)?.entryTypeMetadata.term ?? 'entry';
```

Comment:
This code will return "entry" in the error case, when the passed event didn't reference an entry.  I know we don't like exposing the term "entry", but the user shouldn't see it if everything is working right, and, IMHO, it's better than a blank space.